be/src/exec/hdfs_scanner.cpp

```diff
@@ -19,8 +19,36 @@
 #include "io/compressed_input_stream.h"
 #include "io/shared_buffered_input_stream.h"
 #include "util/compression/stream_compression.h"
+#include "util/lru_cache.h"
 namespace starrocks {
 
+FooterCache::FooterCache(size_t capacity) : _cache(capacity) {}
+
+bool FooterCache::get(const std::string& key, std::string* value) {
+    CacheKey k(key);
+    Cache::Handle* h = _cache.lookup(k);
+    if (h == nullptr) return false;
+    Slice s = _cache.value_slice(h);
+    value->resize(s.size);
+    memcpy(value->data(), s.data, s.size);
```

Comment:
I guess it's hard to maintain.

Plus, this footer cache is supposed to support orc format.

this optimization is for reducing IO instead of memcpy/deserialization.